### ISSUE TITLE
Transforming collision models

### ISSUE THREAD

**Issue by Yuri6037** [dimforge/rapier#243]
What's the best way to scale a collision model such as a compound collision, a triangle mesh collision model, etc.? Ideally this would be good to apply arbitary vector based scaling to any collision model without having to rebuild it. I think Havok has a way to do this considering Source Engine.

Can you apply shearing to the different collision models?

**Comment by frankieali** [comment]
+1 for scaling collision models. I'm attempting to replace AmmoJS with Rapier.js and `setLocalScaling` on convex hulls is a feature I use and would like to see available in rapier.

**Comment by frankieali** [comment]
Any chance of getting uniform scaling soon? It would be nice to be able to scale a rigid bodies without having to rebuild it. Lots of use cases for this.

**Comment by sebcrozet** [comment]
@frankieali Have you conciserded building the scaled shape and assinging it to the existing collider using `Collider.setShape`?

**Comment by Kek5chen** [comment]
But if a shape is being rescaled once every frame that would be absolutely horrible to rebuild the whole trimesh. Transformations would be so much cheaper and better, and easier to use. This is honestly a must have imo.

https://docs.unity3d.com/ScriptReference/BoxCollider-size.html
https://docs.unity3d.com/ScriptReference/CapsuleCollider.html

**Comment by frankieali** [comment]
@sebcrozet Yes, I did try rebuilding the scaled shape. It works well enough for one time scale changes. For example, I can create a new object with a scale param, and the scale is applied only once. However, as @Kek5chen mentions above, I can not easily animate objects that are also being scaled. For instance, a loot bag that, when triggered, causes multiple items to pop out from it and scale up (As seen in Legend of Zelda games). In this case there are multiple colliders all being scaled each step.

**Comment by JarSeal** [comment]
+1 For this feature for now I can do with changing the shape to another (one time change).

**Comment by Calandiel** [comment]
Is scaling the trimesh manually the intended workflow here?

I've been using rapier in a game that renders a lot of vegetation (which involves thousands of slightly differently scaled tree models) - naively cloning the collider to scale it for each individual tree (as they all have more or less unique scales) ends up using literally gigabytes of RAM, which makes the library seem less than ideal ^^'
